File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int n = nums.size();
        unordered_map<int,int> prefixCount(n);
        int ans = 0;
        int prefix = 0;
        prefixCount[prefix] = 1;
        for (int i = 0; i < n; i++) {
            prefix += nums[i];
            ans += prefixCount[prefix - k];
            prefixCount[prefix]++;
        }
        return ans;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
from collections import defaultdict

class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        seen = defaultdict(int)
        seen[0] = 1
        prefix = 0
        count = 0
        
        for n in nums:
            prefix += n
            if prefix - k in seen:
                count += seen[prefix - k]
            seen[prefix] += 1
        return count